web/src/features/panels/zone/ZoneHeader.tsx

```diff
@@ -49,13 +51,56 @@ export function ZoneHeader({
   const fossilFuel =
     (isConsumption ? fossilFuelRatio : fossilFuelRatioProduction) ?? null;
 
+  const [outageData, setOutageData] = useState(null);
+  const fetchData = async () => {
+    const id = '1C7zULJugzjFj3zw98WZYzBhEIHNYzTtTLFdTcVdzqzc';
+    const gid = '0';
+    const url = 'https://docs.google.com/spreadsheets/d/'+id+'/gviz/tq?tqx=out:json&tq&gid='+gid;
+    const responseText = await (
+      await fetch(url)
+    ).text();
+    const table = JSON.parse(responseText.match(/(?<=.*\().*(?=\);)/s)[0]).table;
+    const matching = table.rows.find(d => d.c[0].v === zoneId);
+    if (matching) {
+      setOutageData({
+        'zoneId': matching.c[0].v,
+        'message': matching.c[1].v,
+        'issueId': matching.c[2].v,
+      });
+    } else {
+      setOutageData(null);
+    }
+  };
+  useEffect(() => {
+    fetchData();
+  }, [zoneId]);
```

Comment:
Would it not be better to do this in the app-backend and return it as a part of the zone data?

Right now we are introducing an additional dependency on google docs being accessible to every client device and we are making extra network calls. By moving it we can replace thousands of network calls on client device (on a aggregated level) with just one call in the app-backend and there is no requirement for the client device to be able to access google docs.

Reply:
Absolutely! This was just a proof of concept. You are absolutely right.